Clinical-Genomics/scout
# Improve LDAP documentation and include a practical example with Futurama LDAP demo auth server

- Use lib flask-ldapconn instead of flask_ldap3_login> to handle ldap authentication - to better handle connection errors (fix #3056)
- Improve documentation and fixing minor things in the login setup documentation (fix #2938)
- Step-to-step description on how to set up the LDAP login system in Scout using the demo LDAP server [docker-test-openldap](https://github.com/rroemhild/docker-test-openldap)

**How to test**:
1. Start a dockerized LDAP demo server containing the characters of futurama:
```
docker pull rroemhild/test-openldap
docker run --rm -p 10389:10389 -p 10636:10636 rroemhild/test-openldap
```
2. Choose a futurama character from those available on the server. Note that each character has an email and a password, Those will be used for the login in Scout. For instance let's take [Leela](https://github.com/rroemhild/docker-test-openldap#cnturanga-leelaoupeopledcplanetexpressdccom)
![Leela](https://static.wikia.nocookie.net/enfuturama/images/2/2f/Turanga_Leela_-_Official_Character_Promo.jpg/revision/latest/scale-to-width-down/250?cb=20211007092538)
3. Let's save a User for Leela in Scout:
`scout --demo load user -i cust000 -u "Leela Turanga" -m leela@planetexpress.com -id leela@planetexpress.com`
4. Modify the demo config in scout to be able to connect to the LDAP server, by adding the following lines:
```
LDAP_HOST = "localhost"
LDAP_PORT = 10389
LDAP_BASE_DN = "dc=planetexpress,dc=com"
LDAP_USER_LOGIN_ATTR = "mail"
LDAP_USE_SSL = False
LDAP_USE_TLS = True
```
5. Start the demo scout server and log in as Leela
6. Bonus test: try to connect as another Futurama character that is not already present in the Scout database.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
